### ISSUE TITLE
document 'space', '-' and (') flags in [makefilename]

### ISSUE THREAD

**Issue by porres** [pure-data/pddp#263]
source https://en.wikipedia.org/wiki/Printf

I just found we can use an escaped space as a flag, which prepends a space for positive signed-numeric types. I will investigate more things throughly and edit and include more details here.

edit: I now also found out how the '-' flag works, which justifies to the left with the width parameter! I have tested and it works with [makefilename] as well.

I tested the apostrophe and it doesn't work...

**Comment by porres** [comment]
well, I guess the space is the only undocumented flag that works, I tested the apostrophe and it doesn't work...

**Comment by porres** [comment]
hmmm, found out about the '-' flag, which justifies to the left with the width parameter!

**Comment by porres** [comment]
> I tested the apostrophe and it doesn't work...

Well, I'm also working on externals and I see that the apostrophe flag won't just simply work out of the box. 

First, the apostrophe (') flag in printf-style formatting in C is used to include a locale-specific thousands separator (such as a comma , or a space) in the output of numbers. It is useful when formatting large integers or floating-point numbers for readability. For example, the integer 1234567  becomes: 1,234,567, and a floating point would be 1,234,567.00000

The ' flag works only if the program sets a locale that supports it. By default, the C locale does not enable thousands separators, so you need to use a locale like en_US.UTF-8.

In my external I was able to enable it by doing 

#include <locale.h> 

and then at the new function doing:

setlocale(LC_NUMERIC, "en_US.UTF-8");

I haven't tested altering the code of [makefilename] for this and I should really just open an issue and perhaps a PR for that in the regular Pd respository.

**Comment by porres** [comment]
TLDR, we have 3 undocumented flags, minus (-), space ( ) and apostrophe ('). The latter is also actually unsupported and would need a simple code revision to support it.
